OrbitGl/SamplingReportDataView.cpp

```diff
@@ -249,10 +249,20 @@ void SamplingReportDataView::OnContextMenu(const std::string& action, int menu_i
   }
 }
 
-void SamplingReportDataView::OnSelect(std::optional<int> index) {
-  uint64_t function_address = index.has_value() ? GetSampledFunction(index.value()).absolute_address
-                                                : SamplingReport::kInvalidFunctionAddress;
-  sampling_report_->OnSelectAddress(function_address, tid_);
+void SamplingReportDataView::OnMultiSelect(const std::vector<int>& indices) {
+  size_t num_indices = indices.size();
+  std::vector<uint64_t> addresses(num_indices);
+  for (size_t i = 0; i < num_indices; ++i) {
+    addresses[i] = GetSampledFunction(indices[i]).absolute_address;
+  }
+  sampling_report_->OnSelectAddresses(addresses, tid_);
+
+  if (!is_filtering_or_sorting_) {
+    selected_indices_.clear();
+    for (int row : indices) {
+      selected_indices_.insert(indices_[row]);
+    }
+  }
```

Comment:
I find the is_filtering_or_sorting_ mechanism a bit confusing. Is it in place only to not repopulate the selected_indices_ set? What would happen if we were to repopulate it when filtering or sorting?

Reply:
I agree that the `is_filtering_or_sorting` mechanism is not very straightforward. 

When the user edits the filter or sorts a column, we will redo selections in the treeview to guarantee the selection consistency, i.e.: we are selecting the same visible functions, rather than selecting the same rows, or cleaning the previous selections.
This reselection in the treeview is made according to the user's last selections stored in either `DataView::selected_indices` or `DataView::selected_index`. So when filtering or sorting, we will find the visible selected functions according to "user's last selections" and "the current visible functions in the treeview". 

Suppose we populate the reselection in the treeview to the dataview when filtering or sorting. The only difference compared with not populate is that we change `DataView::selected_indices` or `DataView::selected_index` according to the reselection. Therefore, we lose information about the user's last selection.